### ISSUE TITLE
Bogus prefer_const_constructors lint when using type arguments

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1382]
_From @Hixie on March 15, 2018 0:10_

```dart
class Test<T> { const Test(); }

Test<T> test<T>() => new Test<T>();

void main() {
  test<int>();
}
```
->
```
  lint • Prefer const with constant constructors at test.dart:3:22 • prefer_const_constructors
```
...but that's bogus, because if you actually use a const constructor here, it will fail at runtime since `T` isn't const.

_Copied from original issue: dart-lang/sdk#32544_

**Comment by pq** [comment]
_From @Hixie on September 10, 2018 7:2_

I just ran into this again.

**Comment by a14n** [comment]
I think it has been fixed at some point (See #1219)

**Comment by pq** [comment]
Oh, that's great.  Thanks @a14n !
